clouddriver-oracle-bmcs/src/main/groovy/com/netflix/spinnaker/clouddriver/oraclebmcs/security/OracleBMCSNamedAccountCredentials.groovy

```diff
@@ -0,0 +1,109 @@
+import com.oracle.bmc.objectstorage.ObjectStorageClient
+
+class OracleBMCSNamedAccountCredentials implements AccountCredentials<OracleBMCSCredentials> {
+
+  String cloudProvider = OracleBMCSCloudProvider.ID
+  String name
+  String environment
+  String accountType
+  String compartmentId
+  List<String> requiredGroupMembership = []
+  OracleBMCSCredentials credentials
+  String region
+  List<OracleBMCSRegion> regions
+  ComputeClient computeClient
+  VirtualNetworkClient networkClient
+  ObjectStorageClient objectStorageClient
+  IdentityClient identityClient
+
+  OracleBMCSNamedAccountCredentials(String name, String environment, String accountType, List<String> requiredGroupMembership, String compartmentId) {
+    this.name = name
+    this.environment = environment
+    this.accountType = accountType
+    this.requiredGroupMembership = requiredGroupMembership
+    this.compartmentId = compartmentId
+
+    AuthenticationDetailsProvider provider = new ConfigFileAuthenticationDetailsProvider(this.name)
+    this.computeClient = new ComputeClient(provider)
+    this.computeClient.setRegion(Region.US_PHOENIX_1)
+    this.networkClient = new VirtualNetworkClient(provider)
+    this.networkClient.setRegion(Region.US_PHOENIX_1)
+    this.objectStorageClient = new ObjectStorageClient(provider)
+    this.objectStorageClient.setRegion(Region.US_PHOENIX_1)
+    this.identityClient = new IdentityClient(provider)
+    this.identityClient.setRegion(Region.US_PHOENIX_1)
+    this.region = Region.US_PHOENIX_1.regionId
+    this.regions = [new OracleBMCSRegion(name: Region.US_PHOENIX_1.regionId,
+      availabilityZones: this.identityClient.listAvailabilityDomains(ListAvailabilityDomainsRequest.builder()
+        .compartmentId(this.compartmentId)
+        .build()).items.collect { it.name })]
+
+  }
+
+  static class Builder {
+    String name
+    String environment
+    String accountType
+    OracleBMCSCredentials credentials
+    List<String> requiredGroupMembership = []
+    String compartmentId
+
+    Builder name(String name) {
+      this.name = name
+      return this
+    }
+
+    Builder environment(String environment) {
+      this.environment = environment
+      return this
+    }
+
+    Builder accountType(String accountType) {
+      this.accountType = accountType
+      return this
+    }
+
+    Builder credentials(OracleBMCSCredentials credentials) {
+      this.credentials = credentials
+      return this
+    }
+
+    Builder requiredGroupMembership(List<String> requiredGroupMembership) {
+      this.requiredGroupMembership = requiredGroupMembership
+      return this
+    }
+
+    Builder compartmentID(String compartmentID) {
+      this.compartmentId = compartmentID
+      return this
+    }
+
+    OracleBMCSNamedAccountCredentials build() {
+      return new OracleBMCSNamedAccountCredentials(this.name, this.environment, this.accountType, this.requiredGroupMembership, this.compartmentId)
+    }
+  }
+
+  class OracleBMCSRegion {
```

Comment:
Same as other clouds really. Currently we have 1 Region with 3 Availability Domains. More info here https://docs.us-phoenix-1.oraclecloud.com/Content/General/Concepts/regions.htm